src/explorer.ts

```diff
@@ -117,23 +120,25 @@ class KubernetesContext implements KubernetesObject {
         ];
     }
 
-    getTreeItem(): vscode.TreeItem | Thenable<vscode.TreeItem> {
+    getTreeItem(type: string | undefined): vscode.TreeItem | Thenable<vscode.TreeItem> {
         const treeItem = new vscode.TreeItem(this.id, vscode.TreeItemCollapsibleState.Collapsed);
-        treeItem.contextValue = "vsKubernetes.cluster";
+        treeItem.contextValue = type ? type : KUBERNETES_CLUSTER;
```

Comment:
This definitely feels like it should be based on a `this.type` field - the type is an attribute of the object, not a transient parameter that might vary from method call to method call.